srml/executive/src/lib.rs

```diff
@@ -14,7 +14,55 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-//! Executive: Handles all of the top-level stuff; essentially just executing blocks/extrinsics.
+//! # Executive Module
+//!
+//! The executive module acts as the orchestration layer for the runtime. It dispatches incoming
+//! extrinsic calls to the respective modules in the runtime.
+//!
+//! ## Overview
+//!
+//! The executive module is coupled with the [system module](../srml_system/index.html) and lets the runtime
+//! communicate with the SRML. It is unique from other modules in that it does not incorporate the `decl_event!`,
+//! `decl_storage!`, or `decl_module!` macros.
+//!
+//! The executive module provides functions to:
+//!
+//! - Check transaction validity.
+//! - Initialize a block.
+//! - Apply extrinsics.
+//! - Execute a block.
+//! - Finalize a block.
+//! - Start an off-chain worker.
+//!
+//! ### Implementations
+//!
+//! The executive module provides the following implementations:
+//!
+//! - `ExecuteBlock`: Trait that can be used to execute a block.
```

Comment:
Do we need this section? I think we decided not to mention implementation details in this section of the docs.

Reply:
It's not really describing the logic. I see it as, "avoid coupling with this module if these traits have what you need." Since executive in particular doesn't have a `decl_module!` or `impl Module`, then maybe it's not necessary at all.